### PR TITLE
New release: 2.4

### PR DESCRIPTION
**Pull request by k00ni**
This pull request contains my latest changes. Background information can be found in #106 

**Summary:**
* files in store folder are now PSR-2 compliant
* extended test environment to cover major parts of RDF store code
* extended travis to run tests against **all** major releases of MySQL (5.5+) and MariaDB (10.0+)
* implemented adapter approach, when accessing the database. this means, mysqli is still default and **no** code changes are necessary. But you can also use PDO, which only requires a few changes in your configuration. For details, please have a look at #109 
* improved documentation and infos in README.md
* added cache solution (PDO only) - see #118
* added [document](https://github.com/semsol/arc2/blob/2.4/doc/SPARQL-support.md) reflecting up-to-date information about SPARQL support of the RDF store

~~Latest working CI run: https://travis-ci.org/semsol/arc2/builds/392194249~~

Related Github project with issues: https://github.com/semsol/arc2/projects/1 

@semsol: i already created a new release draft in the releases section, labeled it `2.4`. Can you read it and get back to me, if you find a problem (*or just fix it, if you want*)? After merging this pull request, can you make the new release for the public? That would be great. Thanks!

### CODE REVIEW

File: store/ARC2_StoreDeleteQueryHandler.php [hunk 1]
```diff
@@ -46,9 +46,13 @@ public function runQuery($infos)
         if ($tc && ($this->refs_deleted || (1 == rand(1, 100)))) {
             $this->cleanTableReferences();
         }
+        // TODO What does this rand() call here? remove it and think about a cleaner way
```
Comment (semsol) [inline store/ARC2_StoreDeleteQueryHandler.php]: Yeah, not very clean. It was just based on experience that MyISAM breaks on a regular basis, and repairing the tables once in 100 queries, and cleaning up unused values every 500 queries was a pragmatic hack to keep the store in a working state.

**Comment by k00ni** [comment]
Nice to hear! =)

File: store/ARC2_StoreSelectQueryHandler.php [hunk 2]
```diff
@@ -1320,6 +1320,9 @@ public function detectExpressionValueType($pattern_ids)
         return '';
     }
 
+    /**
+     * @todo not in use, so remove?
```
Comment (semsol) [inline store/ARC2_StoreSelectQueryHandler.php]: The method calls are built dynamically in code. IIRC, they come from stuff like "<", "<", "&&", "||" in FILTERs.

**Comment by k00ni** [comment]
:+1: thanks for merging

**Comment by k00ni** [comment]
Travis run with after the merge fails, unfortunately: https://travis-ci.org/semsol/arc2/builds/415344632

**Pull request merged**